GitUp/Application/AppDelegate.m

```diff
@@ -69,6 +69,38 @@ - (AboutWindowController *)aboutWindowController {
   return _aboutWindowController;
 }
 
+- (WelcomeWindowController *)welcomeWindowController {
+  if (!_welcomeWindowController) {
+    _welcomeWindowController = [[WelcomeWindowController alloc] init];
+    
+    __weak NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+    _welcomeWindowController.model.getUserDefaultsShouldShow = ^BOOL{
+      return [defaults boolForKey:kUserDefaultsKey_ShowWelcomeWindow];
+    };
+    
+    __weak NSDocumentController *documentController = NSDocumentController.sharedDocumentController;
+    _welcomeWindowController.model.getRecentDocuments = ^NSArray<NSURL *> * _Nonnull{
+      return documentController.recentDocumentURLs;
+    };
+    
+    __weak typeof(self) weakSelf = self;
+    _welcomeWindowController.model.configureItem = ^(NSMenuItem * _Nonnull item) {
+      item.target = weakSelf;
+      item.action = @selector(_openDocument:);
+    };
+    
+    _welcomeWindowController.model.openTwitter = ^{
+      [[NSWorkspace sharedWorkspace] openURL:[NSURL URLWithString:kURL_Twitter]];
+    };
+    
+    __weak AppDelegate *appDelegate = self;
+    _welcomeWindowController.model.viewIssues = ^{
+      [appDelegate viewIssues:nil];
```

Comment:
Here as well, just let the window's button action talk to the window controller and open a URL. No need to delegate this information back to the AppDelegate.